### DISCUSSION TITLE
How to pass an env var to an un-owned build script?

### DISCUSSION THREAD

**Discussion by dk333333** [bazelbuild/rules_rust#2532] [Q&A]
Hi,

When adding `pyo3` as a dep to a `rust_library`, it runs its own build script. I would like to configure `PYO3_PYTHON` https://pyo3.rs/v0.20.3/building_and_distribution.html?highlight=interpreter#configuring-the-python-version.

Does rules_rust have a way to do this? For `rust_library`, I only see `rustc_env`. [cargo_build_script](https://bazelbuild.github.io/rules_rust/flatten.html#cargo_build_script) does have `build_script_env`, but in this case I do not own the build script / did not make a `cargo_build_script`. 

Also, using `PYO3_PYTHON=... bazel run ...` does not seem to propregate the env to the build script, is this supposed to be the case?

**Comment by illicitonion** [comment]
Assuming you're using crate_universe, you should be able to use https://bazelbuild.github.io/rules_rust/crate_universe.html#crate.annotation-build_script_env - https://github.com/bazelbuild/rules_rust/blob/main/examples/crate_universe/WORKSPACE.bazel has some examples of setting up (other) annotations when using crate_universe

**Comment by akesling** [comment]
I was just doing this myself.  In my case, the environment variable needed to be set on the `pyo3-ffi` crate and so I added the following to my `crates_repository` in my `WORKSPACE` file:

```starlark
crates_repository(
    ...
    annotations = {
        "pyo3-ffi": [crate.annotation(
            build_script_env = {
                "PYO3_CROSS_PYTHON_VERSION": "3.11",
            },
        )],
    },
    ...
)
```
